Docs 5865 ch required application updates hn (#225)
mulesoft/docs-runtime-manager: modules/ROOT/pages/cloudhub-app-runtime-version-updates.adoc

= Updating the Runtime Version for CloudHub Apps 
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: cloudhub, runtime manager, arm, runtime version, mule, updates,

image:logo-cloud-active.png[link="/runtime-manager/deployment-strategies", title="CloudHub"]
image:logo-hybrid-disabled.png[link="/runtime-manager/deployment-strategies", title="Hybrid Deployment"]
image:logo-server-disabled.png[link="/runtime-manager/deployment-strategies", title="Anypoint Platform Private Cloud Edition"]

When you deploy an application to CloudHub, you have access to the validated OS updates, security patches, and bug fixes that MuleSoft provides regularly.

> You can keep your CloudHub applications up to date by manually updating your apps at your convenience or you can let MuleSoft apply the updates automatically.

Reviewer: ```suggestion
You can keep your CloudHub applications up to date by manually updating your apps at your convenience, or you can let MuleSoft apply the updates automatically.
```

When a new runtime version is available for your CloudHub application, the *Applications* page displays a banner and the *Update Required* tab lists the apps that require updates:

.The screenshot shows (*1*) the banner and (*2*) the *Update Required* tab on the *Applications* page.
image::cloudhub-app-update-req-page.png[Applications page with banner and Update Required tab]

If no apps require updates, no banner appears and the *Update Required* tab is empty.

== Update the App Manually 

> The *Update Required* tab indicates one or more apps require an update. You can schedule the update to occur at your convenience.

Reviewer: ```suggestion
The *Update Required* tab indicates that one or more apps require an update. You can schedule the update to occur at your convenience.
```

Before updating apps in the Production environment, update them in the Sandbox environment and run required tests.

To update an app to the latest security patch:

> . In Runtime Manager, in the menu on the left, click *Applications*.

Reviewer: ```suggestion
. In the Runtime Manager left menu, click *Applications*.
```
Tempted to say "In the Runtime Manager navigation menu, click...".

. Click the *Update Required* tab.

> . Click the app status to display the details pane on the right.

Reviewer: How does one click the app status? Is there a label or icon or something in the UI?

. Click the *Manage Application* button to display the *Settings* page for the app.
+
If the app requires an update, *Update available* appears in the *Runtime version* field:
+
.The screenshot shows (*1*) the *Update available* button and (*2*) the *Runtime version* menu on the *Settings* page.
image::cloudhub-app-update-available-setting.png[Update available on the application Settings page]

. Select the latest version from the *Runtime version* menu.
+

> Although the version number of the update might be the same as the installed version, the date distinguishes the update.

Reviewer: ```suggestion
Although the version number of the update might be the same as the installed version, a later date identifies the update.
```

. If you want to see details about the version, click the *Read release notes* link.
. Click the *Apply Changes* button to update the app. 

CloudHub updates the app with the version you selected and displays a message indicating success or failure.

If the app update was successful, the app no longer appears in the *Update Required* tab on the *Applications* page.

> If the app update was unsuccessful, the app remains in the *Update Required* tab.

Reviewer: Is there text following or a link topic that addresses how to troubleshoot this?

= Updating the Runtime Version for CloudHub Apps 
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: cloudhub, runtime manager, arm, runtime version, mule, updates,

image:logo-cloud-active.png[link="/runtime-manager/deployment-strategies", title="CloudHub"]
image:logo-hybrid-disabled.png[link="/runtime-manager/deployment-strategies", title="Hybrid Deployment"]
image:logo-server-disabled.png[link="/runtime-manager/deployment-strategies", title="Anypoint Platform Private Cloud Edition"]

To provide a safe cloud environment for applications and data, MuleSoft regularly scans apps deployed on CloudHub and identifies apps that require updates to the CloudHub runtimes.
These updates provide the latest OS and security patches. To keep your apps up-to-date and secure, apply the updates when they become available.

You can schedule and update your apps manually at your convenience or you can wait for MuleSoft to apply the updates automatically.

MuleSoft updates CloudHub apps automatically, based on the environment, on this schedule:

[%header,cols="2"]
|===
|Environment |App Update Schedule 
|*Production*|Every three months

> |*Sandbox*|Once every month

Reviewer 2: @vinayakjangam Please advise the updated schedule

+
.The screenshot shows (*1*) the banner and (*2*) the *Update Required* tab on the *Applications* page.
image::cloudhub-app-update-req-page.png[Applications page with banner and Update Required tab]

* The *Settings* page for the app that requires updates displays a banner and the *Update Required* button appears in the *Runtime version* field:
+
.The screenshot shows (*1*) the banner and (*2*) the *Update Required* button on the *Settings* page.
image::cloudhub-app-settings-update-req.png[Settings page with banner and Update Required button]

// Is this a future enhancement?
// The banner on the Applications page indicates the date when the updates will be applied automatically.

If no apps require updates, no banner appears and the *Update Required* tab is empty.

Because these updates contain required security patches, you cannot roll back these updates once applied.

== Update the App Manually 

The *Update Required* tab indicates one or more apps require an update to Mule. You can schedule the update to occur at your convenience.

Before updating apps in the Production environment, update them in the Sandbox environment and run required tests.

> To update the Mule version for an app:

Reviewer 2: It's not updating the Mule Version of the app, it's more like going to the latest update of the Mule Version they are currently on
```suggestion
To update an app to the latest security patch:
```

= Updating the Runtime Version for CloudHub Apps 
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: cloudhub, runtime manager, arm, runtime version, mule, updates,

image:logo-cloud-active.png[link="/runtime-manager/deployment-strategies", title="CloudHub"]
image:logo-hybrid-disabled.png[link="/runtime-manager/deployment-strategies", title="Hybrid Deployment"]
image:logo-server-disabled.png[link="/runtime-manager/deployment-strategies", title="Anypoint Platform Private Cloud Edition"]

> To provide a safe cloud environment for applications and data, MuleSoft regularly scans apps deployed on CloudHub and identifies apps that require updates to the CloudHub runtimes.

Reviewer 3: @hannanelson We are revisiting the strategy for these updates. So for now, I would like to change this section (up until the snapshot) to reflect the current status and update it again in a couple of week when we have a better schedule.

Author: @evinaku - what is the current status?

These updates provide the latest OS and security patches.
To keep your apps up-to-date and secure, apply the updates when they become available.

Because these updates contain required security patches, you cannot roll back these updates once applied.

You can schedule and update your apps manually at your convenience or you can wait for MuleSoft to apply the updates automatically.

MuleSoft updates CloudHub apps automatically, based on the environment, on this schedule:

[%header,cols="2"]
|===
|Environment |App Update Schedule 
|*Production*|Every three months
|*Sandbox*|Once every month
|===

When a new version is available and CloudHub application updates are required, the *Applications* page displays a banner and the *Update Required* tab lists the apps that require updates:

> 

Reviewer 3: @hannanelson Can the section above this be done away with for now and replaced with something like this:

"When you run an application on CloudHub, you get access to validated OS updates, security patches and bug fixes on a regular basis. You can keep you applications on CloudHub applications upto date by manually take the patch updates at your convenience or let the platform update these applications for you regularly.

When a new date patch version of the mule runtime your application is running is available the
Applications page displays a banner and the Update Required tab lists the apps that require
updates:"

We will revisit this once the strategy is crystalized.
